File: 이것이 취업을 위한 코딩 테스트다/정렬/선택_정렬/선택_정렬.cpp
#include <iostream>
using namespace std;

int n = 10;
int arr[] = {7, 5, 9, 0, 3, 1, 6, 2, 4, 8};

int main() {
  for (int i = 0; i < 10; i++) {
    int minIndex = i;

    for (int j = i + 1; j < 10; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    swap(arr[i], arr[minIndex]);
  }

  for (int i = 0; i < 10; i++) {
    cout << arr[i] << ' ';
  }

  return 0;
}

File: 이것이 취업을 위한 코딩 테스트다/정렬/선택_정렬/선택_정렬.java
public class 선택_정렬 {
  public static void main(String[] args) {
    int arr[] = {9, 8, 7, 6, 5, 4, 3, 2, 1, 0};

    for (int i = 0; i < 10; i++) {
      int minIndex = i;

      for (int j = i + 1; j < 10; j++) {
        if (arr[minIndex] > arr[j]) {
          minIndex = j;
        }
      }

      int temp = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = temp;
    }

    for (int num: arr) {
      System.out.print(num + " ");
    }
  }
}
